testcases/OpTestInterrupt.py

```diff
@@ -0,0 +1,189 @@
+
+
+class OpTestInterrupt(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.platform = conf.platform()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        self.c = self.cv_SYSTEM.console
+        self.con = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+    
+    def xive_test(self):
+        '''
+        This function will do below steps 
+        1.Enable kernel logging, basically generate traffic
+        2.Enable console traffic by printing the 
+        processes/tasks to the console
+        '''
+        self.c.run_command("echo 10 > /proc/sys/kernel/printk")
+        try:
+            i = 1
+            while i <= 10: 
+                self.c.run_command("echo t > /proc/sysrq-trigger", timeout=600)
+                i = i+1
+        except(KernelOOPS, KernelKdump):
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.c.run_command("echo $?")
+
+class XiveTesting(OpTestInterrupt):
+    '''
+    1.This function will check xive is configured properly or not
+    2.check for system reboot with Xive=off
+    3.check for system reboot with Xive=on
+    '''
+    def setup_test(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.con.run_command("uname -a")
+        result = self.con.run_command("grep -i --color=never xive /boot/config-`uname -r`")
+        for line in result:
+            temp = ''.join(self.con.run_command("echo %s | cut -d '=' -f 2" % line))
+            if temp != 'y':
+                self.fail("xive is not configured properly")
+        res = self.con.run_command("cat /etc/os-release | grep NAME | head -1")
+        if 'SLES' in res[0].strip():
+            self.distro = 'SLES'
+            self.con.run_command_ignore_fail("sed -i -e 's/xive=.* / /' -e 's/xive=.*/\"/' /etc/default/grub")
+            self.con.run_command("sed -i '/GRUB_CMDLINE_LINUX_DEFAULT/s/\"$/ xive=off\"/' /etc/default/grub")
+            self.con.run_command("grub2-mkconfig -o /boot/grub2/grub.cfg")
+            self.con.run_command("sync;sleep 10")
+            self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+            self.con.run_command("cat /proc/cmdline")
+            self.con.run_command_ignore_fail("cat /proc/interrupts | grep -Eai 'xive'")
+            self.con.run_command("sed -i 's/xive=.*/xive=on\"/' /etc/default/grub")
+            self.con.run_command("grub2-mkconfig -o /boot/grub2/grub.cfg")
+            self.con.run_command("sync;sleep 10")
+            self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+            self.con.run_command("cat /proc/cmdline")
+        elif 'Red Hat' in res[0].strip():
+            self.distro = 'RHEL'
+            self.rhel_grub('off')
+            self.rhel_grub('on')
+
+    def rhel_grub(self, param):
```

Comment:
Better rhel_grub_update or some such function name .. which reflects what you are doing?